### PR TITLE
Add vim intro and exercise to linux-intro class

### PR DESCRIPTION
**Pull request by yifatmakias**

### CODE REVIEW

File: docs/course_materials/linux_intro.html [hunk 1]
```diff
@@ -515,6 +515,34 @@ <h3><b>clear</b></h3>
           <section data-markdown="exercises/01_Linux/class_exercise_2.md" data-separator="^\n\n\n" data-separator-vertical="^\n\n">
           </section>
         </section>
+        <section>
+          <section>
+            <h3><b>Vim</b></h3>
+            <ul>
+              <li>Vim is an editor to create or edit a text file.</li>
+              <li>There are two modes in vim. One is the command mode and another is the insert mode.</li>
```
Comment (shiramax) [inline docs/course_materials/linux_intro.html:523]: > 
```suggestion
              <li>There are two modes in vim: command mode and  insert mode.</li>
```

File: docs/course_materials/linux_intro.html [hunk 2]
```diff
@@ -515,6 +515,34 @@ <h3><b>clear</b></h3>
...
+                <li>In the command mode, user can move around the file, delete text, etc.</li>
+                <li>In the insert mode, user can insert text.</li>
+              </ul>
+            </ul>
+          </section>
+          <section>
+            <h3><b>Usefull vim commands</b></h3>
+            <ul>
+              <li>i - Enter insert mode.</li>
+              <li>esc - Exit insert mode.</li>
```
Comment (shiramax) [inline docs/course_materials/linux_intro.html:534]: a few more useful things I think we should add:
jump to the end of the file, 
jump to the beginning of the file. 
search words
go to a specific line. 
@yifatmakias can you add those?

**Review by shiramax — CHANGES_REQUESTED** [review]

File: docs/course_materials/linux_intro.html (hunk 2 above)
Comment (yifatmakias) [reply docs/course_materials/linux_intro.html:534]: Yes, I will add those :)

File: docs/course_materials/linux_intro.html [hunk 3]
```diff
@@ -515,6 +515,42 @@ <h3><b>clear</b></h3>
...
+            <ul>
+              <li>Vim is an editor to create or edit a text file.</li>
+              <li>There are two modes in vim: command mode and insert mode.</li>
+              <ul>
+                <li>In the command mode, user can move around the file, delete text, etc.</li>
+                <li>In the insert mode, user can insert text.</li>
+              </ul>
+            </ul>
+          </section>
+          <section>
```
Comment (shiramax) [inline docs/course_materials/linux_intro.html:530]: Thanks @yifatmakias 
can you add another slide for how to create a file with vim and how to open an existing file with vim.

File: docs/course_materials/linux_intro.html [hunk 4]
```diff
@@ -515,6 +515,42 @@ <h3><b>clear</b></h3>
...
+            <h3><b>Usefull vim commands</b></h3>
+            <ul>
+              <li>i - Enter insert mode</li>
+              <li>esc - Exit insert mode</li>
+              <li>:w - Save file</li>
+              <li>:wq - Save and close file</li>
+              <li>:q - Close file</li>
```
Comment (shiramax) [inline docs/course_materials/linux_intro.html:537]: @yifatmakias can you add wq! ?

File: docs/course_materials/exercises/01_Linux/class_exercise_3.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+### Let's Practice Together \#3
+
+- Open a new file with the vim editor
+- Enter insert mode
+- Write 4 sentences about your self
```
Comment (shiramax) [inline docs/course_materials/exercises/01_Linux/class_exercise_3.md:5]: it should be 4 sentences in 4 different lines.
maybe we can tell them to paste a text, for example: 
add this text to the fie: 
"Roses are red,
Violets are blue,
Sugar is sweet,
And so are you.

Roses are red,
Violets are blue,
White wine costs less,
Than dinner for two.

Roses are red,
But violets aren’t blue,
They’re purple, you dope,
Now go get a clue.
"

and then you can  another question and ask then to search "Roses" , 
a bonus question is to replace all "Roses" with "watermelons"

**Review by shiramax — CHANGES_REQUESTED** [review]

**Review by shiramax — CHANGES_REQUESTED** [review]
Thanks @yifatmakias it looks good now, 
It could be great if the class exercise will be in one file.
we can create a file with this text. and the students to CP the file to there own file. 
that way, the entire exercise could be in the same file.

also, the name of the class exercise is 31 on purpose ?

**Comment by yifatmakias** [comment]
@shiramax Yes because I created 2 files of exercise 3 I named one 31 and the other 32.
I can change it back to one file. Did you mean to change it to one file and remove the text from it?

**Review by shiramax — APPROVED** [review]
thanks @yifatmakias !

**Pull request merged**